site-src/geps/gep-957.md

```diff
@@ -0,0 +1,82 @@
+# GEP-957: Destination Port Matching
+
+* Issue: [#957](https://github.com/kubernetes-sigs/gateway-api/issues/957)
+* Status: Provisional
+
+## TLDR
+
+Add a new `port` field to ParentRef to support port matching in Routes.
+
+## Goals
+
+* Support port matching in routes based on the destination port number of the
+  request.
+
+## Non-Goals
+
+* Support port matching based on named port.
+
+## Introduction
+
+Port matching is a common service mesh use case where traffic policies/rules
+need to be applied to traffic to certain destination ports. For ingress, while
+the API today already supports attaching a route to a specific listener, it may
+be useful to support attaching routes to listener(s) on a specified port. For
+example, some users may only want to expose their application on ports 80 or 443
+on the Gateway.
+
+## API
+
+The proposal is to add a new field `Port` to `ParentRef`:
+
+```go
+type ParentRef struct {
+  ...
+  // Port is the network port this resource targets. It can be interpreted
+  // differently based on the type of parent resource:
+  //
+  // Gateway: All listeners listening on the specified port that also support
+  // this kind of Route. When both Port and SectionName are specified, the
+  // name and port of the selected listener must match both specified values.
+  //
+  // Implementations MAY choose to support other parent resources.
+  // If that is the case, they MUST clearly document how Port is
+  // interpreted.
+  //
+  // When unspecified, this will reference the entire resource.
+  // For the purpose of status, an attachment is considered successful as
+  // long as the parent resource accepts it partially. For example, Gateway
+  // listeners can restrict which Routes can attach to them by Route kind,
+  // namespace, or hostname. If 1 of 2 Gateway listeners accept attachment from
+  // the referencing Route, the Route MUST be considered successfully
+  // attached. If no Gateway listeners accept attachment from this Route, the
+  // Route MUST be considered detached from the Gateway.
+  //
+  // Support: Core
+  //
+  // +optional
+  Port *PortNumber `json:"port,omitempty"`
+  ...
+}
+```
+
+## Alternatives
+### 1. Use SectionName in ParentRef for port matching
+Port matching can be supported if SectionName accepts port numbers in addition
+to listener names. This approach results in a less explicit API for mesh use
+cases. For example, it's less inituitive to create a route with
+`ParentRef.SectionName` set to `443` to express `route all traffic whose
+destination port is 443 to ...`. It also complicates the validation on
+SectionName in order to differentiate between a listener name and a port number.
```

Comment:
IIUC the only reason that section name isn't sufficient is where the parent is some resource that is not a Gateway. It would be good to call that out here because it's pretty important IMHO.

Reply:
Rephrased this section. Please let me know if it is more clear now.